realm/realm-library/src/main/java/io/realm/CompactOnLaunchCallback.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2017 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm;
+
+/**
+ * CompactOnLaunchCallback interface is used to determine if it should be compacted before being returned to the user.
+ */
+
+public interface CompactOnLaunchCallback {
+
+    /**
+     * This method determines if it should be compacted before returned to the user. It is passed the total file size
```

Comment:
if it -> if the Realm file should be compacted before being opened and returned to the user.